server/src/main/java/org/opensearch/cluster/routing/OperationRouting.java

```diff
@@ -56,59 +55,52 @@
 import java.util.Set;
 import java.util.stream.Collectors;
 
-import static org.opensearch.common.Booleans.parseBoolean;
-
 public class OperationRouting {
 
     public static final Setting<Boolean> USE_ADAPTIVE_REPLICA_SELECTION_SETTING =
             Setting.boolSetting("cluster.routing.use_adaptive_replica_selection", true,
                     Setting.Property.Dynamic, Setting.Property.NodeScope);
 
-    private static final DeprecationLogger deprecationLogger = DeprecationLogger.getLogger(OperationRouting.class);
-    private static final String IGNORE_AWARENESS_ATTRIBUTES_PROPERTY = "opensearch.search.ignore_awareness_attributes";
-    static final String IGNORE_AWARENESS_ATTRIBUTES_DEPRECATION_MESSAGE =
-        "searches will not be routed based on awareness attributes starting in version 8.0.0; " +
-            "to opt into this behaviour now please set the system property [" + IGNORE_AWARENESS_ATTRIBUTES_PROPERTY + "] to [true]";
-
-    private List<String> awarenessAttributes;
-    private boolean useAdaptiveReplicaSelection;
+    public static final String IGNORE_AWARENESS_ATTRIBUTES = "cluster.search.ignore_awareness_attributes";
+    public static final Setting<Boolean> IGNORE_AWARENESS_ATTRIBUTES_SETTING = Setting.boolSetting(IGNORE_AWARENESS_ATTRIBUTES,
+        true, Setting.Property.Dynamic, Setting.Property.NodeScope);
```

Comment:
@nknize @AmiStrn will this be considered backward compatibility breaking? This is doing 2 things
1. Getting rid of the deprecated system property based override mechanism to toggle routing preferences and replacin git with a cluster setting 
2. Changing the default value from false (of now removed system property) to true (of new cluster setting) - end result is ARS gets enabled everywhere

A nice side effect is bugs like https://github.com/elastic/elasticsearch/issues/53054 will get addressed with default=true

Reply:
Yes, it is breaking imo. 
for (1) this is fine, I think we should be doing this. Backporting to 1.1 can be done with a better deprecation message.
for (2) I would say the same, however, I noticed some bugs in the ARS implementation were addressed in Elasticsearch since the fork. These should be checked before making this the default.
one example of a bug we forked to opensearch: https://github.com/elastic/elasticsearch/issues/65838
(this is a mix up in the formula implementation so I wouldn't worry about any copyright stuff on that one).

Reply:
@AmiStrn are you saying we should change the default in this PR back to old behavior before merging?

Reply:
The previous deprecation message was "searches will not be routed based on awareness attributes starting in version 8.0.0" which tells me that the intent was for awareness attribute based routing was planned to be removed in version 8.0.0 (i.e. OpenSearch 2.0). It appears to me that this change keeps the awareness attribute-based routing feature (though sets the default to disabled) and removes the deprecation message. Should we just be removing awareness attributed-based routing in OpenSearch 2.0 all together?

Reply:
@andrross The idea is to keep the setting to have a mechanism to enable it in future if needed and also monitor the effectiveness before removing the setting. I agree that it is a breaking change which is why its a good idea to have the setting to toggle before completely removing it

Reply:
If the setting is temporary while we monitor the effectiveness of the new default, then shouldn't there be a deprecation message if a user configures `cluster.search.ignore_awareness_attributes` since it will go away in the future?

Reply:
To be clear, I think we should really consider removing the setting. We're confident enough to make the new behavior the default and the setting was deprecated in the previous major version. However, I will defer to the opinion of folks with more knowledge of this specific feature.

Reply:
@andrross  Based on initial tests using rally the results looks promising. But we want to keep a setting to control the feature to toggle it for couple of releases with enabled as default. This will help to see if it is causing any regression with the new behavior or not and have control on it. We can add the deprecation message in later releases when we feel comfortable with this default value and plan to remove the setting completely. 

As far as existing deprecation message is concerned I would think it anyways doesn't make sense from opensearch perspective because of version 8.0.0. We can choose to deprecate it in later releases of opensearch rather than sticking with elastic behavior.